graphql_compiler/compiler/helpers.py

```diff
@@ -183,8 +211,28 @@ def validate_marked_location(location):
         raise GraphQLCompilationError(u'Cannot mark location at a field: {}'.format(location))
 
 
+@six.add_metaclass(ABCMeta)
```

Comment:
I don't think I understand why this is necessary.

Reply:
This simply specifies the contract for what it means for something to be a "real" location object, whether folded or not. `ABCMeta` ensures that all of the contract's methods have to be implemented for a class to be able to be instantiated.